src/Rings/MPolyQuo.jl

```diff
@@ -947,3 +947,59 @@ function dim(a::MPolyQuoIdeal)
   return a.dim
 end
 #######################################################
+
+
+################################################################################
+#
+#  Minimalizing a set of subalgebra generators in graded case
+#
+################################################################################
+@doc Markdown.doc"""
+    minimal_subalgebra_generators(V::Vector{S}) where S <: Union{MPolyElem{U}, MPolyQuoElem{U}} where U
+
+Given a vector `V` collecting homogeneous elements of a graded multivariate polynomial ring or quotient ring, say `R`, return a minimal subset of these elements which still generate the subalgebra of `R` generated by all elements collected in `V`.
+
+NOTE: The conditions on `V` and `R` are automatically checked.
+
+# Examples
+```jldoctest
+julia> R, (x,y,z) =  GradedPolynomialRing(QQ, ["x", "y", "z"])
+(Multivariate Polynomial Ring in x, y, z over Rational Field graded by
+  x -> [1]
+  y -> [1]
+  z -> [1], MPolyElem_dec{fmpq, fmpq_mpoly}[x, y, z])
+
+julia> V = [x,y,x^2+y^2]
+3-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ x
+ y
+ x^2 + y^2
+
+julia> minimal_subalgebra_generators(V)
+2-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ y
+ x
+```
+"""
+function minimal_subalgebra_generators(V::Vector{S}) where S <: Union{MPolyElem{U}, MPolyQuoElem{U}} where U
+    @assert all(x->parent(x) == parent(V[1]), V)
+    if !((typeof(parent(V[1])) <: Oscar.MPolyRing_dec) && (isgraded(parent(V[1]))))
+       throw(ArgumentError("The base ring must be graded."))
+    end
+    if !(all(ishomogeneous, V))
+       throw(ArgumentError("The input data is not homogeneous."))
+    end
+    W = filter(x -> true, V)
```

Comment:
This seems to be an inefficient way to make a copy of V?
```suggestion
    W = copy(V)
```
Alternatively, use `sort` (which makes and returns a copy of its input) instead of `sort!`

Reply:
@fingolfin: Question: I understand, that if I use copy and then change W, I will change V as well. That is, after running the function, V has changed. It should be different if I use filter. Correct?

Reply:
No, `copy` makes a copy, just like `filter` (or, for that matter, `sort` without the `!`), so if you work with that copy, then `V` does not change.

There should never be a reason to write `filter(x->true, V)` :-)

Reply:
O.K. I agree